test/BigDecimal.ts

```diff
@@ -0,0 +1,118 @@
+import { deepStrictEqual } from "effect-test/util"
+import * as BigDecimal from "effect/BigDecimal"
+import { pipe } from "effect/Function"
+import * as Option from "effect/Option"
+
+describe.concurrent("BigDecimal", () => {
+  it("sign", () => {
+    assert.deepStrictEqual(BigDecimal.sign(BigDecimal.make(-5n)), -1)
+    assert.deepStrictEqual(BigDecimal.sign(BigDecimal.make(0n)), 0)
+    assert.deepStrictEqual(BigDecimal.sign(BigDecimal.make(5n)), 1)
+  })
+
+  it("isBigDecimal", () => {
+    expect(BigDecimal.isBigDecimal(BigDecimal.make(1n))).toEqual(true)
+    expect(BigDecimal.isBigDecimal(1)).toEqual(false)
+    expect(BigDecimal.isBigDecimal("a")).toEqual(false)
+    expect(BigDecimal.isBigDecimal(true)).toEqual(false)
+  })
+
+  it("sum", () => {
+    deepStrictEqual(pipe(BigDecimal.make(1n), BigDecimal.sum(BigDecimal.make(2n))), BigDecimal.make(3n))
```

Comment:
Will need to add some more test coverage for each operator with different scales.

Reply:
Oh yeah these are just the ones I copied over so far. I am going to copy the test cases from the bigdecimal.rs rust crate.